### DISCUSSION TITLE
Generated C files?

### DISCUSSION THREAD

**Discussion by ghost** [NVIDIA/open-gpu-kernel-modules#408] [Q&A]
Hello,

I see that https://github.com/NVIDIA/open-gpu-kernel-modules/tree/main/src/nvidia/generated are generated files,
is it possible to have these tools to generate the C corresponding files?

Thank you.

**Comment by mtijanic** [comment]
Hi there! There's a few different things that generate source files, but most of the ones at that link are done by the NVOC tool. Please read the explanation of that tool as given here: https://github.com/NVIDIA/open-gpu-kernel-modules/issues/100#issuecomment-1124898947

Additionally, most of the "SDK" files, such as [these ones](https://github.com/NVIDIA/open-gpu-kernel-modules/tree/main/src/common/sdk/nvidia/inc/ctrl), are generated by a different tool, which was explained here: https://github.com/NVIDIA/open-gpu-kernel-modules/discussions/372#discussioncomment-3653089

Everything else is either using off-the-shelf tools (e.g. protobuf: `*_pb.{c,h}` files), or one-off perl scripts that we have as part of our build flow.

As explained in the other threads, we have opted to publish already generated files as that greatly simplifies the build flow for anyone using these modules. In hindsight, it would have been better if the codebase relied less on code generation, but these things become a necessary evil when dealing with as many platforms as these modules do.

If you don't mind, could you please explain _why_ you'd want these tools? It might help us figure out how to proceed pragmatically.
